src/com/untamedears/realisticbiomes/GrowthConfig.java

```diff
@@ -261,103 +238,65 @@ public boolean isPersistent() {
-		// environment multiplier.
-		if(isGreenhouseEnabled && ( environmentMultiplier < greenhouseRate ) && ( block.getLightFromBlocks() == (MAX_LIGHT_INTENSITY - 1) ) ) {
-			// make sure it's a glowstone/lamp
+		if (isGreenhouseEnabled && block.getLightFromBlocks() == (MAX_LIGHT_INTENSITY - 1)) {
 			for( Vector vec : adjacentBlocks ) {
 				Material mat = block.getLocation().add(vec).getBlock().getType();
 				if( mat == Material.GLOWSTONE || mat == Material.REDSTONE_LAMP_ON ) {
-					environmentMultiplier = greenhouseRate;
+					rate *= greenhouseRate;
 					break;
 				}
 			}
 		}
-		
-		rate *= environmentMultiplier;
-		
-		// if the plant will be effected by irrigation, check if the block is at the correct level and
-		// check if nearby blocks are water blocks in river biomes
-		if (notIrrigatedMultiplier != 1.0) {
-			// determine if the block is near a water block in a river biome
-			boolean irrigated = false;
-			for( Vector vec : waterCheckBlocks ) {
-				Block waterBlock = block.getLocation().add(vec).getBlock();
-				Material mat = waterBlock.getType();
-				Biome biome = waterBlock.getBiome();
-				if((biome == Biome.RIVER || biome == Biome.FROZEN_RIVER) && (mat == Material.STATIONARY_WATER || mat == Material.WATER)) {
-					irrigated = true;
-					break;
-				}
+		else {
+			int sunlightIntensity;
+			if (block.getType().isTransparent()) {
+				sunlightIntensity = block.getLightFromSky();
+			} else {
+				sunlightIntensity = block.getRelative(BlockFace.UP).getLightFromSky();
 			}
-			
-			if (!irrigated)
-				rate *= notIrrigatedMultiplier;
-		}
-		
-		// check the depth of the required 'soil' and add a bonus
-		float soilBonus = 0.0f;
-		Block newBlock = block.getRelative(0,-soilLayerOffset,0);	
-		int soilCount = 0;
-		while (soilCount < soilMaxLayers) {
-			if (newBlock == null || !newBlock.getType().equals(soilMaterial)) {
-				break;
+			if (needsSunlight) {
+				rate *= Math.pow((sunlightIntensity / MAX_LIGHT_INTENSITY), 3.0);
 			}
-			if (soilData != -1 && newBlock.getData() != soilData) {
-				break;
+			// apply multiplier if the sunlight is not at maximum
+			if (sunlightIntensity < MAX_LIGHT_INTENSITY) {
+				rate *= notFullSunlightMultiplier;
 			}
-				
-			soilBonus += soilBonusPerLevel;
-			
-			newBlock = newBlock.getRelative(0, -1, 0);
-			soilCount++;
 		}
-		rate *= (1.0 + soilBonus);
-		
-		return rate;
+		// check the depth of the required 'soil' and add a bonus
+				float soilBonus = 0.0f;
+				Block newBlock = block.getRelative(0,-soilLayerOffset,0);	
+				int soilCount = 0;
+				while (soilCount < soilMaxLayers) {
+					if (newBlock == null || !newBlock.getType().equals(soilMaterial)) {
+						break;
+					}
+					if (soilData != -1 && newBlock.getData() != soilData) {
+						break;
+					}
+						
+					soilBonus += soilBonusPerLevel;
+					
+					newBlock = newBlock.getRelative(0, -1, 0);
+					soilCount++;
+				}
+				rate *= (1.0 + soilBonus);
+				return rate;		

```

Comment:
Yeah, I'll fix the indentation, didn't even notice that. I dont think it needs to be wrapped into an additional if, because soilMaxLayers will already be 0 by default. This part was identical in the previously used method.